src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml

```diff
@@ -6,7 +6,7 @@
       <method signature="System.Boolean IsEnabled(System.Diagnostics.Tracing.EventLevel,System.Diagnostics.Tracing.EventKeywords)" body="stub" value="false" />
       <method signature="System.Boolean IsEnabled(System.Diagnostics.Tracing.EventLevel,System.Diagnostics.Tracing.EventKeywords,System.Diagnostics.Tracing.EventChannel)" body="stub" value="false" />
     </type>
-    <type fullname="System.Globalization.GlobalizationMode">
+    <type fullname="System.Globalization.GlobalizationMode/Settings">
```

Comment:
This effectively removes `Invariant = true` dependencies trimming. I think it'd be better to substitute at the current level.

Reply:
> This effectively removes Invariant = true dependencies trimming.

Why?

Reply:
I tested it and the dead branches still get trimmed. Why do you think this removes that?

Also, you can’t substitute this at the higher level because then the Settings class wouldn’t be touched and it’s cctor won’t be run.

And there were concerns in the issue about putting the cctor on GlobalizationMode itself because calling UseNls would force it to run. So splitting the cctor off in a separate class was proposed.

Reply:
> Also, you can’t substitute this at the higher level because then the Settings class wouldn’t be touched and it’s cctor won’t be run.

I think that's what we want for `Invariant = true`. I think Settings for that mode is redundant.

Reply:
I'm not sure how to resolve this comment, plus address all the concerns raised in https://github.com/dotnet/runtime/issues/49073#issuecomment-794550693. Can you elaborate on what you want here?

Do you want me to split the ILLink.Substitutions.xml apart so `Invariant = true` and `Invariant = false` look different? For example:

```xml
    <type fullname="System.Globalization.GlobalizationMode">
      <method signature="System.Boolean get_Invariant()" body="stub" value="true" feature="System.Globalization.Invariant" featurevalue="true" />
    </type>
    <type fullname="System.Globalization.GlobalizationMode/Settings">
      <method signature="System.Boolean get_Invariant()" body="stub" value="false" feature="System.Globalization.Invariant" featurevalue="false" />
    </type>
```

That doesn't seem like an ideal situation IMO.

I don't think your original concern is a problem. Can you explain what are the disadvantages to the current approach I am using?